Nautilus/Patchers/VehicleUpgradesPatcher.cs

```diff
@@ -35,28 +34,59 @@ internal static void Patch(Harmony harmony)
     [HarmonyPatch(typeof(Vehicle), nameof(Vehicle.OnUpgradeModuleChange))]
     private static void OnModuleChangeDelegate(Vehicle __instance, int slotID, TechType techType, bool added)
     {
-        if (!VehicleUpgradeModules.TryGetValue(techType, out var prefab))
-            return;
+        if(__instance is Exosuit)
+        {
+            if (!ExosuitUpgradeModules.TryGetValue(techType, out var prefab))
+                return;
 
-        if (!prefab.TryGetGadget(out UpgradeModuleGadget moduleGadget))
-            return;
+            if (!prefab.TryGetGadget(out UpgradeModuleGadget moduleGadget))
+                return;
```

Comment:
Could this be moved to the beginning of the patch?

Reply:
No, I'm making a distinction between ExosuitUpgradeModules and VehicleUpgradeModules